src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorJsonDataPayload.java

```diff
@@ -484,6 +484,32 @@ else if (queueDataAggregationMethod
     return jsonObject;
   }
 
+  /**
+   * Gets the boolean value average from the specified array of values.
+   *
+   * <p>If the number of true values is greater than the number of false values, {@link
+   * Boolean#TRUE} is returned. Otherwise, {@link Boolean#FALSE} is returned.
+   *
+   * @param seriesValues the array of values to get the average from
+   * @return the boolean value average
+   * @since 1.0.1
+   */
+  private static synchronized Object getBooleanValueAvg(Object[] seriesValues) {
+    // Count true and false values
+    int trueCount = 0;
+    int falseCount = 0;
+    for (int i = 0; i < seriesValues.length; i++) {
+      if (Boolean.TRUE.equals(seriesValues[i])) {
+        trueCount++;
+      } else if (Boolean.FALSE.equals(seriesValues[i])) {
+        falseCount++;
+      }
+    }
+
+    // Return average (true if true count is greater than false count, false otherwise)
+    return trueCount > falseCount ? Boolean.TRUE : Boolean.FALSE;
```

Comment:
The comment was improved, although `false otherwise` was also intended to cover ties.

Additionally, I brought the logic in line with rounding up at 0.5 (trueCount == falseCount), as mentioned.